[WIP] refactor narrowing, add test, run narrow before copying

WillForan opened the pull request:
also address issue #27 and could supersede pull request #28

I pulled out the`meta` length check into it's own function and added a test. I also moved the `narrow_participants` call before temporary files are copied.

The test is only checking narrowing works, not any down stream effects. And I'm worried I've likely overlooked some important considerations. Please feel free to [ruthlessly reject](https://www.jeffgeerling.com/blog/2022/burden-open-source-maintainer)

WillForan committed:
[BUG] only one el in BIDS.participants.meta

WillForan committed:
[WIP] narrow_participants.m and test, run before copy

Remi-Gau commented:
will try to review in the coming days

makes me think that this kind of thing should probably be delegated to bids-matlab which is an update and better version of spm_bids

Remi-Gau commented:
but already thanks for adding tests!!!

Remi-Gau commented:
Note to self:
- circle CI should run on PR
- add code linting
- use matlab github actions to run tests

Remi-Gau commented on narrow_participants.m:
```diff
@@ -0,0 +1,25 @@
+function BIDS = narrow_participants(BIDS, labels)
+% NARROW_PARTICIPANTS - restrict BIDS struct to only requested labels
+%
```
```suggestion
% 
```

Remi-Gau committed:
Update narrow_participants.m

Remi-Gau commented on test_narrow_participants.m:
Do you think you could rewrite this test to be used directly by matlab testing framework?

Ideally with setup and teardown functions.

https://nl.mathworks.com/help/matlab/matlab_prog/write-test-using-setup-and-teardown-functions.html

If you don't have the bandwidth, let me know and I can give it a go

Remi-Gau commented:
Seems that we have a path issue when testing it:

https://app.circleci.com/pipelines/github/bids-apps/SPM/44/workflows/dd6d3670-b56f-419b-bb88-b8b6ec8f8021/jobs/220

Remi-Gau commented:
> Seems that we have a path issue when testing it:
> 
> https://app.circleci.com/pipelines/github/bids-apps/SPM/44/workflows/dd6d3670-b56f-419b-bb88-b8b6ec8f8021/jobs/220

that's because the new m file you have added is not copied into the container: 

https://github.com/bids-apps/SPM/blob/68e5b93de860d841714ddb411d1675b65779c8cb/Dockerfile#L41

Remi-Gau committed:
'precommit_update'

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/pre-commit/pre-commit-hooks: v4.5.0 → v4.6.0](https://github.com/pre-commit/pre-commit-hooks/compare/v4.5.0...v4.6.0)

Remi-Gau committed:
autoupdate_precommit

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/pre-commit/pre-commit-hooks: v4.6.0 → v5.0.0](https://github.com/pre-commit/pre-commit-hooks/compare/v4.6.0...v5.0.0)
- [github.com/hadolint/hadolint: v2.13.0-beta → v2.13.1-beta](https://github.com/hadolint/hadolint/compare/v2.13.0-beta...v2.13.1-beta)

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/pre-commit/pre-commit-hooks: v5.0.0 → v6.0.0](https://github.com/pre-commit/pre-commit-hooks/compare/v5.0.0...v6.0.0)
- [github.com/hadolint/hadolint: v2.13.1-beta → v2.14.0](https://github.com/hadolint/hadolint/compare/v2.13.1-beta...v2.14.0)

WillForan committed:
fix(narrow): skip participant field if empty

WillForan committed:
fix(Dockerfile): track narrow_participants.m

WillForan committed:
wip: global BIDS on pipeline

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci
